gibbon-compiler/src/Gibbon/HaskellFrontend.hs

```diff
@@ -179,6 +181,40 @@ type TopTyEnv = TyEnv TyScheme
 
 type TypeSynEnv = M.Map TyCon Ty0
 
+-- ========================================================
+
+getImportMeta :: [H.ImportDecl a] -> M.Map Var (Var, Bool, Maybe [Var])
```

Comment:
I may have mentioned it elsewhere but I want to reiterate: this Map is a core data structure in your work, and that's why It should have a name introduced either as `type` or a `newtype` (the former is easier of a change but long-term less sustainable).